test/e2e/searchindex_test.go

```diff
@@ -0,0 +1,226 @@
+			searchIndexConfig = &akov2.AtlasSearchIndexConfig{
+				TypeMeta: metav1.TypeMeta{
+					Kind:       "AtlasSearchIndexConfig",
+					APIVersion: "atlas.mongodb.com/v1",
+				},
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "test-search-index-config",
+					Namespace: testData.InitialDeployments[0].Namespace,
+				},
+				Spec: akov2.AtlasSearchIndexConfigSpec{
+					Analyzer:       pointer.MakePtr("lucene.standard"),
+					SearchAnalyzer: pointer.MakePtr("lucene.standard"),
+				},
+			}
+			Expect(testData.K8SClient.Create(testData.Context, searchIndexConfig)).To(Succeed())
+		})
+
+		By("Creating one search index, type: SEARCH", func() {
+			Eventually(func(g Gomega) {
+				g.Expect(testData.K8SClient.Get(testData.Context, types.NamespacedName{
+					Name:      testData.InitialDeployments[0].Name,
+					Namespace: testData.InitialDeployments[0].Namespace,
+				}, testData.InitialDeployments[0])).To(Succeed())
+
+				searchIndexesToCreate := []akov2.SearchIndex{
+					{
+						Name:           "test-search-index",
+						Type:           atlasdeployment.IndexTypeSearch,
+						DBName:         DBTraining,
+						CollectionName: DBTrainingCollectionRoutes,
+						Search: &akov2.Search{
+							Mappings: &akov2.Mappings{Dynamic: pointer.MakePtr(true)},
+							SearchConfigurationRef: common.ResourceRefNamespaced{
+								Name:      searchIndexConfig.GetName(),
+								Namespace: searchIndexConfig.GetNamespace(),
+							},
+						},
+					},
+				}
+				testData.InitialDeployments[0].Spec.DeploymentSpec.SearchIndexes = searchIndexesToCreate
+				g.Expect(testData.K8SClient.Update(testData.Context, testData.InitialDeployments[0])).To(Succeed())
+			}).WithPolling(10 * time.Second).WithTimeout(10 * time.Minute).Should(Succeed())
+
+			Eventually(func(g Gomega) {
+				g.Expect(testData.K8SClient.Get(testData.Context, types.NamespacedName{
+					Name:      testData.InitialDeployments[0].Name,
+					Namespace: testData.InitialDeployments[0].Namespace,
+				}, testData.InitialDeployments[0])).To(Succeed())
+				g.Expect(len(testData.InitialDeployments[0].Status.SearchIndexes)).To(BeEquivalentTo(1))
+				g.Expect(testData.InitialDeployments[0].Status.SearchIndexes[0].Status).To(BeEquivalentTo(status.SearchIndexStatusReady))
+			}).WithPolling(10 * time.Second).WithTimeout(20 * time.Minute).Should(Succeed())
+		})
+
+		By("Creating one search index, type: VECTOR SEARCH", func() {
+			Eventually(func(g Gomega) {
+				g.Expect(testData.K8SClient.Get(testData.Context, types.NamespacedName{
+					Name:      testData.InitialDeployments[0].Name,
+					Namespace: testData.InitialDeployments[0].Namespace,
+				}, testData.InitialDeployments[0])).To(Succeed())
+
+				vectorSearchIndexToCreate := akov2.SearchIndex{
+					Name:           "test-vector-search-index",
+					Type:           atlasdeployment.IndexTypeVector,
+					DBName:         DBTraining,
+					CollectionName: DBTrainingCollectionGrades,
+					VectorSearch:   &akov2.VectorSearch{},
```

Comment:
I believe the e2e test failing is missing some configuration here:

```shell
https://cloud-qa.mongodb.com/api/atlas/v2/groups/6634bc7dee5df86c35d9f7c2/clusters/search-nodes-test/fts/indexes POST: HTTP 400 Bad Request (Error code: "MISSING_ATTRIBUTE") Detail: The required attribute [fields] was not specified. Reason: Bad Request. Params: [[fields]]
```

Will try to figure out a simple fields config that would work.